ClosedXML_Tests/Excel/CalcEngine/FunctionsTests.cs

```diff
@@ -658,5 +659,131 @@ public void TestOmittedParameters()
                 Assert.AreEqual(2, value);
             }
         }
+
+        [Test]
+        public void NumberValue()
+        {
+            Object actual;
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123.45\", \".\", \",\")");
+            Assert.AreEqual(123.45, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123,45\", \",\", \".\")");
+            Assert.AreEqual(123.45, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123.456.789,01\", \",\", \".\")");
+            Assert.AreEqual(123456789.01, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1,234.56\", \".\", \",\")");
+            Assert.AreEqual(1234.56, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1.234,56\", \",\", \".\")");
+            Assert.AreEqual(1234.56, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"123.45\")");
+            Assert.AreEqual(123.45, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1,234.56\")");
+            Assert.AreEqual(1234.56, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"+1\")");
+            Assert.AreEqual(1, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"+1.23\")");
+            Assert.AreEqual(1.23, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"-1\")");
+            Assert.AreEqual(-1, actual);
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"-1.23\")");
+            Assert.AreEqual(-1.23, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\".1\")");
+            Assert.AreEqual(0.1, actual);
+
+            actual = XLWorkbook.EvaluateExpr("NUMBERVALUE(\"1.234567890E+34\")");
```

Comment:
BTW, `1.234567890E+307` is processed the same way by your code and by Excel. But `1.234567890E+308` fails in Excel and succeeds in your implementation.